src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,198 @@
+        default:
+            break;
+        }
+    }
+
+    /**
+     * Handles the Deadline command and its details. It checks if the details are valid and how many of the date and/or
+     * time is/are specified. If only the Date is specified, the Deadline task will only have a date. If only the Time
+     * is specified, the Deadline Task will be set to today's Date and the specified Time. If both the Date and Time are
+     * specified, the Deadline task will be set to the specified Date and Time. Then the Deadline task will
+     * be added into the TaskList.
+     *
+     * @param reply    Details of Deadline command
+     * @param taskList Overall TaskList of all the Tasks
+     * @throws DukeException If details of Deadline is invalid(i.e. insufficient arguments, argument in incorrect
+     *                       format)
+     */
+    public static void handleDeadline(String reply, TaskList taskList) throws DukeException {
+        String[] taskReplyArr = reply.split("/by ");
+        if (taskReplyArr.length < 2) {
+            Ui.throwDeadlineInputError();
+        }
+        String[] taskInstrArr = taskReplyArr[0].split(" ");
+        try {
+            String task = taskInstrArr[1];
+            for (int i = 2; i < taskInstrArr.length; i++) {
+                task += " " + taskInstrArr[i];
+            }
+            String timeDate = taskReplyArr[1];
+            String[] timeDateArr = timeDate.split(" ");
+            if (timeDateArr.length == 2) {
+                if (DATE_VALIDATOR.isValidDate(timeDateArr[0]) && TIME_VALIDATOR.isValidTime(timeDateArr[1])) {
+                    LocalDate formattedDate = LocalDate.parse(timeDateArr[0], DATE_FORMATTER);
+                    LocalTime formattedTime = LocalTime.parse(timeDateArr[1], TIME_FORMATTER);
+
+                    Deadline deadLine = new Deadline(task, formattedDate, formattedTime, false);
+                    taskList.addTask(deadLine);
+                    Ui.showTaskAdded(deadLine, taskList);
+                } else {
+                    Ui.throwDeadlineInputError();
+                }
+            } else if (timeDateArr.length == 1) {
+                if (DATE_VALIDATOR.isValidDate(timeDateArr[0])) {
+                    LocalDate formattedDate = LocalDate.parse(timeDateArr[0], DATE_FORMATTER);
+                    Deadline deadLine = new Deadline(task, formattedDate, false);
+                    taskList.addTask(deadLine);
+                    Ui.showTaskAdded(deadLine, taskList);
+                } else if (TIME_VALIDATOR.isValidTime(timeDateArr[0])) {
+                    LocalTime formattedTime = LocalTime.parse(timeDateArr[0], TIME_FORMATTER);
+                    Deadline deadLine = new Deadline(task, LocalDate.now(), formattedTime, false);
+                    taskList.addTask(deadLine);
+                    Ui.showTaskAdded(deadLine, taskList);
+                } else {
+                    Ui.throwDeadlineInputError();
+                }
+
+            }
```

Comment:
- **Parsing** of user input should be done in the **Parser** class. Then it can be assumed that the parameters received here are valid, just need to assert.
- Alot of repeated code here! Try to abstract them into separate functions
- Same goes for `handleEvent()`